mlflow/protos/model_registry.proto

```diff
@@ -848,3 +920,149 @@ message GetModelVersionByAlias {
     optional ModelVersion model_version = 1;
   }
 }
+
+// Webhooks
+message Webhook {
+  // Unique name for the webhook.
+  optional string name = 1;
+
+  // Timestamp recorded when this ``webhook`` was created.
+  optional int64 creation_timestamp = 2;
+
+  // Timestamp recorded when metadata for this ``webhook`` was last updated.
+  optional int64 last_updated_timestamp = 3;
+
+  // User that created this ``webhook``
+  // NOTE: this field is not currently returned.
+  optional string user_id = 4;
+
+  // url of the ``webhook``.
+  optional string url = 5;
+
+  // Event type of the ``webhook``.
+  optional string event_trigger = 6;
+
+  // The key of the event trigger for the ``webhook``.
+  optional string key = 7;
+
+  // The value of the event trigger for the ``webhook``.
+  optional string value = 8;
+
+  // The headers to include in the request
+  map<string, string> headers = 9;
+
+  // The payload to include in the request
+  map<string, string> payload = 10;
+
+  // Description of this ``webhook``.
+  optional string description = 11;
+}
+
+message CreateWebhook {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Create webhook under this name
+  optional string name = 1 [(validate_required) = true];
+
+  // url of the ``webhook``.
+  optional string url = 2;
+
+  // Event type of the ``webhook``.
+  optional string event_trigger = 3;
+
+  // The key of the event trigger for the ``webhook``.
+  optional string key = 4;
+
+  // The value of the event trigger for the ``webhook``.
+  optional string value = 5;
+
+  // The headers to include in the request
+  map<string, string> headers = 6;
+
+  // The payload to include in the request
+  map<string, string> payload = 7;
+
+  // Description of this ``webhook``.
+  optional string description = 8;
+
+  message Response {
+    optional Webhook webhook = 1;
+  }
+}
+
+message RenameWebhook {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Webhook unique name identifier.
+  optional string name = 1 [(validate_required) = true];
+
+  // If provided, updates the name for this ``webhook``.
+  optional string new_name = 2;
+
+  message Response {
+    optional Webhook webhook = 1;
+  }
+}
+
+message UpdateWebhook {
```

Comment:
Are we following Databricks webhook APIs? If so  its UpdateWebhook actually updates the status instead of description https://docs.databricks.com/api/workspace/modelregistry/updatewebhook

Reply:
Yes, the OSS model registry webhook can be implemented independently from Databricks one.